agent/src/main/java/com/teamscale/jacoco/agent/Agent.java

```diff
@@ -160,10 +161,13 @@ private void dumpReportUnsafe() {
 
 		try (Benchmark benchmark = new Benchmark("Generating the XML report")) {
 			FileSystemUtils.ensureParentDirectoryExists(outputPath.toFile());
-			coverageFile = generator.convert(dump, outputPath);
+			coverageFile = generator.convert(dump, outputPath, false);
 		} catch (IOException e) {
 			logger.error("Converting binary dump to XML failed", e);
 			return;
+		} catch (EmptyReportException e) {
+			logger.warn("No coverage was collected. Please check the exception message for more details.", e);
```

Comment:
I don't think you need to explicitly tell the user to read the exception message. It's printed in the log right under the warning